content/client-server-api/modules/threading.md

```diff
@@ -0,0 +1,201 @@
+when `is_falling_back` is `true`.
+
+{{% boxes/note %}}
+Clients which are acutely aware of threads (they do not render threads, but are otherwise
+aware of the feature existing in the spec) can treat rich replies to an event with a `rel_type`
+of `m.thread` as a threaded reply, for conversation continuity on the threaded client's side.
+
+To do this, copy the `event_id` (thread root) from the event being replied to, add the
+`m.in_reply_to` metadata, and add `is_falling_back: true` to `m.relates_to`.
+{{% /boxes/note %}}
+
+#### Replies within threads
+
+In the [fallback for unthreaded clients](#fallback-for-unthreaded-clients) section, a new
+`is_falling_back` flag is added to `m.relates_to`. This flag defaults to `false` when not
+provided, which also allows a threaded message to contain a reply itself.
+
+Aside from `is_falling_back` being `false` (or not specified), the fallback for unthreaded
+clients is used to create a reply within a thread: clients should render the event accordingly.
+
+#### Server behaviour
+
+##### Validation of `m.thread` relationships
+
+Servers SHOULD reject client requests which attempt to start a thread off an event with a
+`rel_type`. If the client attempts to target an event which already has an `m.thread`,
+`m.reference`, or any other `rel_type` then it should receive a HTTP 400 error response
+with appropriate error message, as per the [standard error response](#standard-error-response)
+structure.
+
+{{% boxes/note %}}
+A specific error code is not currently available for this case: servers should use `M_UNKNOWN`
+alongside the HTTP 400 status code.
+{{% /boxes/note %}}
+
+##### Server-side aggreagtion of `m.thread` relationships
+
+Given threads always reference the thread root, an event can have multiple "child" events which
+then form the thread itself. These events should be [aggregated](#aggregations) by the server.
+
+The aggregation for threads includes some information about the user's participation in the thread,
+the approximate number of events in the thread (as known to the server), and the most recent event
+in the thread (topologically). This is then bundled into the event as `m.thread`:
+
+```json
+{
+  "event_id": "$root_event",
+  // irrelevant fields not shown
+  "unsigned": {
+    "m.relations": {
+      "m.thread": {
+        "latest_event": {
+          // A serialized copy of the latest event in the thread.
+          // Some fields are not shown here for brevity.
+          "event_id": "$message",
+          "sender": "@alice:example.org",
+          "room_id": "!room:example.org",
+          "type": "m.room.message",
+          "content": {
+            "msgtype": "m.text",
+            "body": "Woo! Threads!"
+          }
+        },
+        "count": 7,
+        "current_user_participated": true
+      }
+    }
+  }
+}
+```
+
+`latest_event` is the most recent event (topologically to the server) in the thread sent by an
+un-[ignored user](#ignoring-users).
```

Comment:
I'm not sure it needs to be called out, but the bundled `latest_event` should be serialized using the same format as the event itself.

Reply:
This is done via the spec and examples implicitly: the other weird formats aren't in the spec anymore, as that's a legacy implementation detail in some clients and servers.

Reply:
Oh, I hadn't realized those were no longer in the spec. Please ignore me then!